esphome/components/modbus_controller/__init__.py

```diff
@@ -126,13 +130,42 @@
 
 _LOGGER = logging.getLogger(__name__)
 
+
+def validate_address(register_set_key: str):
+    register_set = CORE.data.setdefault(DOMAIN, {}).setdefault(register_set_key, set())
+
+    def validator(register: dict):
```

Comment:
Looking over this again, I think this needs to be inside the actual validator
```suggestion
    def validator(register: dict):
        register_set = CORE.data.setdefault(DOMAIN, {}).setdefault(register_set_key, set())```

Reply:
I had to modify this proposed fix to keep the linter happy, but now it is inside the validator